src/states/node.rs

```diff
@@ -976,32 +1140,65 @@ impl Node {
 
     fn handle_node_identify(&mut self, public_id: PublicId, peer_id: PeerId) -> Evented<()> {
         let mut result = Evented::empty();
-        debug!("{:?} Handling NodeIdentify from {:?}.",
-               self,
-               public_id.name());
-
-        if let Some((name, _)) = self.sent_network_name_to {
-            if name == *public_id.name() {
-                self.sent_network_name_to = None;
+        debug!("{:?} Handling NodeIdentify from {:?}.", self, public_id.name());
+        match self.peer_mgr.check_candidate(&public_id, &peer_id) {
+            Ok(Some((tunnel, seed))) => {
+                if tunnel {
```

Comment:
You could get rid of the `if` and have two match arms instead: `Ok(Some((true, seed)))`, and with `false`.